nomad/config.go

```diff
@@ -472,22 +464,24 @@ func (c *Config) Copy() *Config {
 
 // ConsulServiceIdentity returns the workload identity to be used for accessing
 // the Consul API to register and manage Consul services.
-func (c *Config) ConsulServiceIdentity() *structs.WorkloadIdentity {
-	if c.ConsulConfig == nil {
+func (c *Config) ConsulServiceIdentity(cluster string) *structs.WorkloadIdentity {
+	conf := c.ConsulConfigs[cluster]
```

Comment:
Would it be worth to have a `GetConsulConfigs` function like its done in the client side?

Reply:
After this PR, the only callers that de-index `ConsulConfigs` are here in `nomad/config.go`, so it doesn't really get us any new abstractions to do that. The client-side version does some logic around ex. whether Vault is enabled.